pkgs/build-support/writers/default.nix

```diff
@@ -22,7 +22,21 @@ rec {
       inherit interpreter;
       contentPath = content;
     }) ''
-      echo "#! $interpreter" > $out
+      # On darwin a script cannot be used as an interpreter in a shebang but
+      # there doesn't seem to be a limit to the size of shebang and multiple
+      # arguments to the interpreter are allowed.
+      if [[ -n "${toString pkgs.stdenvNoCC.isDarwin}" ]] && isScript $interpreter
+      then
+        wrapperInterpreter=$(head -1 "$interpreter" | tail -c+3)
+        # This should work as long as wrapperInterpreter is a shell, which is
+        # the for programs wrapped with makeWrapper, like
```

Comment:
```suggestion
        # for programs wrapped with makeWrapper, like
```

Reply:
fixed in 194985b9bbd09e9a58506dc935aa2c0109263ade